inferentia/qa/setup_test_enviroment_and_test.sh

```diff
@@ -0,0 +1,134 @@
+export PYTHON_BACKEND_REPO_TAG=${DEFAULT_REPO_TAG}
+export CHECKSUM_REPOAGENT_REPO_TAG=${DEFAULT_REPO_TAG}
+export TRITON_SERVER_BRANCH_NAME=${TRITON_SERVER_BRANCH_NAME:=${DEFAULT_REPO_TAG}}
+export TRITON_CLIENT_REPO_TAG=${TRITON_CLIENT_REPO_TAG:=${DEFAULT_REPO_TAG}}
+export TRITON_VERSION="2.17.0dev"
+export TRITON_CONTAINER_VERSION="21.12dev"
+export TRITON_UPSTREAM_CONTAINER_VERSION="21.10"
+export BASE_IMAGE=tritonserver
+export SDK_IMAGE=tritonserver_sdk
+export BUILD_IMAGE=tritonserver_build
+export QA_IMAGE=tritonserver_qa
+
+cd ${TRITON_PATH}
+# Clone necessary branches
+rm -rf ${TRITON_PATH}/server
+git clone --single-branch --depth=1 -b ${TRITON_SERVER_BRANCH_NAME} \
+          https://github.com/triton-inference-server/server.git
+echo ${TRITON_VERSION} > server/TRITON_VERSION
+cd ${TRITON_PATH}/server
+git clone --single-branch --depth=1 -b ${TRITON_CLIENT_REPO_TAG} \
+          https://github.com/triton-inference-server/client.git clientrepo
+
+# First set up inferentia and run in detatched mode
+cd ${TRITON_PATH}/python_backend
+chmod 777 ${TRITON_PATH}/python_backend/inferentia/scripts/setup-pre-container.sh
+sudo ${TRITON_PATH}/python_backend/inferentia/scripts/setup-pre-container.sh
+
+# Build container with only python backend 
+cd ${TRITON_PATH}/server
+pip3 install docker
+./build.py --build-dir=/tmp/tritonbuild \
+           --cmake-dir=${TRITON_PATH}/server/build \
+           --version=${TRITON_VERSION} \
+           --container-version=${TRITON_CONTAINER_VERSION} \
+           --enable-logging --enable-stats --enable-tracing \
+           --enable-metrics --enable-gpu-metrics --enable-gpu \
+           --filesystem=gcs --filesystem=azure_storage --filesystem=s3 \
+           --endpoint=http --endpoint=grpc \
+           --repo-tag=common:${TRITON_COMMON_REPO_TAG} \
+           --repo-tag=core:${TRITON_CORE_REPO_TAG} \
+           --repo-tag=backend:${TRITON_BACKEND_REPO_TAG} \
+           --repo-tag=thirdparty:${TRITON_THIRD_PARTY_REPO_TAG} \
+           --backend=identity:${IDENTITY_BACKEND_REPO_TAG} \
+           --backend=python:${PYTHON_BACKEND_REPO_TAG} \
+           --repoagent=checksum:${CHECKSUM_REPOAGENT_REPO_TAG}
+docker tag tritonserver_build "${BUILD_IMAGE}"
+docker tag tritonserver "${BASE_IMAGE}"
+
+# Build docker container for SDK
+docker build -t ${SDK_IMAGE} \
+             -f ${TRITON_PATH}/server/Dockerfile.sdk \
+             --build-arg "TRITON_CLIENT_REPO_SUBDIR=clientrepo" .
+
+# Build QA container
+docker build -t ${QA_IMAGE} \
+                   -f ${TRITON_PATH}/python_backend/inferentia/qa/Dockerfile.QA \
+                   --build-arg "TRITON_PATH=${TRITON_PATH}" \
+                   --build-arg "BASE_IMAGE=${BASE_IMAGE}"   \
+                   --build-arg "BUILD_IMAGE=${BUILD_IMAGE}" \
+                   --build-arg "SDK_IMAGE=${SDK_IMAGE}"     .
+
+export TEST_JSON_REPO=/opt/tritonserver/qa/common/inferentia_perf_analyzer_input_data_json
```

Comment:
Can't we run the single and multiple instance cases in single test script?

Reply:
In the future, we probably want to add more tests for multiple instances that focus on how compute is distributed so it _might_ be cleaner to just have two separate test folders. If the user doesn't care about multiple instances, this way it's also easier to comment out the second test.

Reply:
I think this might be over optimizing. I'll combining the two for now